docs/proposals/scheduling/20220510-load-aware-scheduling.md

```diff
@@ -0,0 +1,137 @@
+5. Estimate the total resource usage of the current node as *P*, *P = A + B + S*
+6. Calculate the score
+    1. If *P >= T*, then *score = 0*
+    2. else *score = (cpu((T - P) * 100 / T * cpuWeight) + memory((T - P) * 100 / T * memoryWeight)) / weightSum*
+
+In step 3, when calculating the estimated value according to **scalingFactor** for Request, we can only process Pods whose QoS is Batch and other types. After supporting the application profile capability in the future, we can use the profile value as the estimated value. Considering that some Pods may not define Request and Limit, a default estimated value needs to be defined, which is currently defined as 250m.
+
+Here is an explanation of why the allocated Pods are also involved in the estimation in step 4. After the Pod is scheduled, it needs to pull the image and mount the disk before it starts. After many seconds have passed since the Pod serving, it will cause the koordlet to report NodeMetrics without missing usages of these Pods. At this time if the scheduler continues to schedule Pods on these nodes, it is likely that the future resource utilization will exceed the threshold. In order to avoid this situation, the plugin implements the Reserve extension point to records the allocated time of the Pod, denoted as *R*, and denote the latest update time of NodeMetric as *U*, then when scheduling the next Pod, the allocated Pods in the current node need to participate in the estimation which meet the conditions *(R > U or U - R > nodeMetricUpdateWindow)*.
+
+If the last update time of the current node's NodeMetric exceeds the configured maximum update time window, the node will get 0 points.
+
+#### Plugin configuration
+
+```go
+
+type LoadAwareSchedulingArgs struct {
+  metav1.TypeMeta
+
+  FilterUnhealthyNodeMetrics         bool                          `json:"filterUnhealthyNodeMetrics,omitempty"`
+  NodeMetricUpdateWindowInSeconds    int64                         `json:"nodeMetricUpdateWindowInSeconds,omitempty"`
+  NodeMetricUpdateMaxWindowInSeconds int64                         `json:"nodeMetricUpdateMaxWindowInSeconds,omitempty"`
+  ResourceWeights                    map[corev1.ResourceName]int64 `json:"resourceWeights,omitempty"`
+  UsageThresholds                    map[corev1.ResourceName]int64 `json:"usageThresholds,omitempty"`
+  EstimatedScalingFactors            map[corev1.ResourceName]int64 `json:"estimatedScalingFactors,omitempty"`
+}
+
+```
+
+- `FilterUnhealthyNodeMetrics` indicates whether to filter nodes where koordlet fails to update NodeMetric.
+- `NodeMetricUpdateWindowInSeconds` indicates the period that koordlet updates NodeMetric, the default is 60s.
+- `NodeMetricUpdateMaxWindowInSeconds` indicates the tolerable maximum period for koordlet to update NodeMetric. If it exceeds, the node is considered abnormal.
```

Comment:
what is the difference with `FilterUnhealthyNodeMetrics`

Reply:
`FilterUnhealthyNodeMetrics` controls the filter behavior, if it is false, `NodeMetricUpdateMaxWindowInSeconds ` X can still be used when scoring

Reply:
ok